extract.sh

```diff
@@ -1,59 +1,88 @@
 #!/bin/bash
-set -evo pipefail
+set -eo pipefail
 
-IMG_TOPICS=("/basler/pylon_camera_node0/image_raw" "/basler/pylon_camera_node1/image_raw")
-PCD_TOPICS=("/azure/points2" "/velodyne/velodyne_points")
-IMU_TOPICS=("/mcu/mcu_imu")
-TEMP_TOPICS=("/mcu/mcu_imu_temp")
-TIME_REF_TOPICS=("/mcu/cameras_ts" "/mcu/lidar_ts")
+SMARTPHONE_VIDEO_DIR="_s10_video_frame"
 
+# Import configuration values
+source extract.conf
 # Activate virtual environment
-source venv/bin/activate;
+source venv/bin/activate
 # Launch roscore in background
 roscore &
 
-for BAG in "${@}";
-  do
-  DATA_DIR="$(basename "$BAG" .bag)"
+BAG=$1
+SMARTPHONE_VIDEO_PATH=$2
 
-  # Create a subdirectory for extraction
-  rm -rf "$DATA_DIR";
-  mkdir "$DATA_DIR";
+DATA_DIR="$(basename "$BAG" .bag)"
 
-  # Image extraction
-  echo "Image data extraction starting..";
-  python2 extract.py --output "$DATA_DIR"\
-    --type image --path "$BAG" --topics "${IMG_TOPICS[@]}";
+# Create a subdirectory for extraction
+rm -rf "$DATA_DIR"
+mkdir "$DATA_DIR"
+
+# Time reference data extraction
+echo "Time reference data extraction starting.."
+python2 extract.py --output "$DATA_DIR"\
+  --type time_ref --path "$BAG" --topics "${TIME_REF_TOPICS[@]}"
+
+# Image extraction
+echo "Image data extraction starting.."
 
-  # IMU data extraction
-  echo "IMU data extraction starting..";
+if [ ${#IMG_TOPICS[@]} -eq 0 ] ; then
+  echo "No image topics provided"
+else
   python2 extract.py --output "$DATA_DIR"\
-    --type imu --path "$BAG" --topics "${IMU_TOPICS[@]}" --temp "${TEMP_TOPICS[@]}";
+  --type image --path "$BAG" --topics "${IMG_TOPICS[@]}"
+fi
 
-  # Time reference data extraction
-  echo "Time reference data extraction starting..";
+# Depth image extraction
+echo "Image data extraction starting.."
+if [ ${#DEPTH_IMG_TOPICS[@]} -eq 0 ] ; then
+  echo "No image topics provided"
+else
   python2 extract.py --output "$DATA_DIR"\
-    --type time_ref --path "$BAG" --topics "${TIME_REF_TOPICS[@]}";
+    --type depth_img --path "$BAG" --topics "${DEPTH_IMG_TOPICS[@]}"
+  # Depth image timestamps alignment
+  for topic in "${DEPTH_IMG_TOPICS[@]}"
```

Comment:
It's more "styleguidish" to follow one style for using bash `for`, `if`, `while` constructions, so choose your lovely one :) for me this one looks more compact

```
if []; then
```

rather than

```
if [] 
  then
```